src/main/java/de/tum/in/www1/artemis/domain/TextBlock.java

```diff
@@ -196,7 +196,7 @@ public String toString() {
 
     @JsonIgnore
     public boolean isAssessable() {
-        return submission.getResult() != null;
+        return submission.getResults() != null;
```

Comment:
This is not enough, you also need to check if the returned list is empty.
I suggest to add a helper method to submission "hasResults" which checks for null and empty.
Again please think about reusable code!